include/wx/dynarray.h

```diff
@@ -508,18 +517,30 @@ class wxBaseObjectArray : private wxBaseArray<T*>
 // avoid clashes between T and symbols defined in wxBaseArray<> scope, e.g. if
 // we didn't do this, we would have compilation problems with arrays of type
 // "Item" (which is also the name of a method in wxBaseArray<>).
-#define WX_DEFINE_TYPEARRAY_WITH_DECL(T, name, base, classdecl)               \
-    typedef wxBaseArray<T> wxBaseArrayFor##name;                              \
-    class name : public wxBaseArrayFor##name                                  \
-    {                                                                         \
-        typedef wxBaseArrayFor##name Base;                                    \
-    public:                                                                   \
-        name() : Base() { }                                                   \
-        explicit name(size_t n) : Base(n) { }                                 \
-        name(size_t n, Base::const_reference v) : Base(n, v) { }              \
-        template <class InputIterator>                                        \
-        name(InputIterator first, InputIterator last) : Base(first, last) { } \
-    }
+#define wxDEFINE_TYPEARRAY_WITH_DECL__BASEARRAY(T, name, base, classdecl)         \
```

Comment:
I'm not really fond of this split macro, but it is the only way I found to avoid duplicating the code. Let me know if you have a better solution for this.